src/gcp/secretManager.ts

```diff
@@ -1,5 +1,22 @@
+import * as iam from "./iam";
+
 import { logLabeledSuccess } from "../utils";
-import * as api from "../api";
+import { FirebaseError } from "../error";
+import { Client } from "../apiv2";
+import { secretManagerOrigin } from "../api";
+
+// Matches projects/{PROJECT}/secrets/{SECRET}
+const SECRET_NAME_REGEX = new RegExp(
+  "projects\\/" +
+    "((?:[0-9]+)|(?:[A-Za-z]+[A-Za-z\\d-]*[A-Za-z\\d]?))\\/" +
+    "secrets\\/" +
+    "([A-Za-z\\d\\-_]+)"
```

Comment:
Can we actually end with a dash or underscore?

Reply:
I wasn't careful enough to check before this comment, but it turns out Secret Manager does allow secrets to end with - or _.